### ISSUE TITLE
Trying to "get_plugin_name" triggers an error

### ISSUE THREAD

**Issue by henriiquecampos** [godotengine/godot#25795]
**Godot version:** 3.1.beta3.official

**OS/device including version:** Solus 3.9999

**Issue description:**

![image](https://user-images.githubusercontent.com/10171059/52561497-b0962700-2de2-11e9-8917-ec491447f966.png)

In the documentation there is a method `EditorPlugin.get_plugin_name() -> String` but trying to call said method returns:

![image](https://user-images.githubusercontent.com/10171059/52561527-cc013200-2de2-11e9-8081-6f6ae82d869f.png)

```
Invalid call. Nonexistent function 'get_plugin_name' in base 'EditorPlugin (plugin.gd)'
```

**Steps to reproduce:**
- Create an EditorPlugin
- In the plugin script call `get_plugin_name`

**Minimal reproduction project:**

**Comment by vnen** [comment]
It's a virtual method, it means you have to implement it in the script.

**Comment by henriiquecampos** [comment]
Ohh, my bad, sorry.

But isn't that weird? I think this method should always return the String with the plugin's folder name, no?

**Comment by Zylann** [comment]
How can `get_plugin_name` be expected to have a script implementation when you need that same name even before activating the plugin, to show it in the list? It's even one of the fields in `plugin.cfg` Oo

**Comment by henriiquecampos** [comment]
Yeah I really don't see a reason for it to be `virtual`. What would be the use case for a script implementation, i.e. override, of this method?

**Comment by Overblob** [comment]
I know a use case: when you want your plugin to be added with the "main editors" ie:
![image](https://user-images.githubusercontent.com/31958866/52900159-dd877700-31f2-11e9-9f32-448bc1b61bf0.png)
the return of `get_plugin_name` is used to display a custom name into the ToolButton.
I don't know if there's another one.

**Comment by Zylann** [comment]
@Overblob if that's the case, it adds to the inconsistency in the way things are named^^"
